rescript/get_rdp_data.py

```diff
@@ -0,0 +1,133 @@
+                            include_species_labels=include_species_labels,
+                            rank_propagation=rank_propagation,
+                            ranks=ranks)
+            # rdp_tax_df.append(taxonomy.view(pd.DataFrame))
+            # rdp_seq_sl.append(rdp_seqs.view(pd.Series))
+            rdp_tax_df.append(taxonomy)
+            rdp_seq_sl.append(rdp_seqs)
+        print('Merging taxonomy ...')
+        merged_rdp_tax, = merge_rdp_taxonomy(data=rdp_tax_df)
+        print('Merging sequences ...')
+        merged_rdp_seqs, = merge_rdp_seqs(data=rdp_seq_sl)
+        print('Writing artifact files...')
+        return merged_rdp_seqs, merged_rdp_tax
+    elif num_seq_files == 1:
+        rdp_seqs = rdp_seq_data[0]
+        rdp_tax, = parse_taxonomy(
+                        rdp_reference_sequences=rdp_seqs,
+                        include_species_labels=include_species_labels,
+                        rank_propagation=rank_propagation,
+                        ranks=ranks)
+        print('Writing artifact files...')
+        return rdp_seqs, rdp_tax
+    elif num_seq_files == 0:
+        raise ValueError('No RDP data to process!')
+    else:
+        raise ValueError('There are more than 2 RDP files! There should '
+                         'be only 1 or 2 files!')
+    # return rdp_seqs, rdp_tax
+
+
+def _assemble_rdp_data_urls(ref_db):
+    '''Generate RDP urls, given database version and reference target.'''
+    # Compile URLs
+    base_url = 'http://rdp.cme.msu.edu/download/'
+
+    if ref_db == 'Archaea+Bacteria':
+        archaea_url = base_url + 'current_Archaea_unaligned.fa.gz'
+        bacteria_url = base_url + 'current_Bacteria_unaligned.fa.gz'
+        return [('archaeal sequences', archaea_url),
+                ('bacterial sequences', bacteria_url)]
+    elif ref_db == 'Fungi':
+        fungi_url = base_url + 'current_Fungi_unaligned.fa.gz'
+        return [('fungal sequences', fungi_url)]
+    else:
+        raise ValueError('Reference database must be either, '
+                         '\'Bacteria+Archaea\' or \'Fungi\'.')
+
+
+def _retrieve_data_from_rdp(queries):
+    '''
+    Download data from RDP, given a list of queries.
+
+    queries: list of tuples of (str, str)
+        (name, urlpath)
+    '''
+    results = []
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        for name, query in queries:
+            print('Retrieving {0} from: {1}'.format(name, query))
+            # grab url
+            destination = os.path.join(tmpdirname, os.path.basename(query))
+            urlretrieve(query, destination)
+
+            # gunzip on demand (RDP releases are gzipped)
+            try:
+                unzipped_destination = os.path.splitext(destination)[0]
+                print('  Unzipping {0}...'.format(name))
+                _gzip_decompress(destination, unzipped_destination)
+                destination = unzipped_destination
+            except OSError:
```

Comment:
what types of errors are encountered here? it seems like you should just error gracefully here if you get an error that would lead to a failed download, instead of failing after attempting to parse the taxonomy later on.